backend/routes/chatRoomRoutes.js

```diff
@@ -0,0 +1,87 @@
+const express = require('express')
+const session = require('express-session')
+const { PrismaClient } = require('../generated/prisma/index.js')
+const isAuthenticated = require('./postRoutes.js')
+const prisma = new PrismaClient()
+const router = express.Router()
+
+
+// Get particular chatroom 
+router.get('/chatrooms/:chatroomId', isAuthenticated, async (req, res) => {
+    try {
+        const chatroomId = parseInt(req.params.chatroomId);
+        const chat = await prisma.chat.findMany({
+            where: {
+                chat_id: chatroomId,
+            },
+            include: {
+                userOne: true,
+                userTwo: true,
+                messages: true,
+            },
+        });
+        if (!chat) {
+            return res.status(404).json({ error: "Chatroom not found" });
+        }
+        res.json(chat);
+    } catch (error) {
+        res.status(500).send("Server error")
+    }
+})
+
+// Get all the chatrooms of current user with the user id
+router.get('/chatrooms/users/:userId', isAuthenticated, async (req, res) => {
+    try {
+        const user_id = parseInt(req.params.userId); 
+        const chats = await prisma.chat.findMany({
```

Comment:
```suggestion
router.get('/chatrooms', isAuthenticated, async (req, res) => {
    try {
        const user_id = parseInt(req.query.userId); 
        const chats = await prisma.chat.findMany({
```

Just a quick reminder about using query params rather than url params here! Here's a quick explainer: https://www.geeksforgeeks.org/node-js/difference-between-req-query-and-req-params-in-express/